backend/src/main/java/codesquard/app/errors/response/ApiResponse.java

```diff
@@ -0,0 +1,43 @@
+package codesquard.app.errors.response;
+
+import org.springframework.http.HttpStatus;
+
+public class ApiResponse<T> {
+
+	private int code;
+	private HttpStatus status;
+	private String message;
+	private T data;
+
+	public ApiResponse(HttpStatus status, String message, T data) {
+		this.code = status.value();
+		this.status = status;
+		this.message = message;
+		this.data = data;
+	}
```

Comment:
메시지를 컨트롤러에서 String으로 받아오면 유지 보수에 안 좋을 수 있다고 생각돼서요!
혹시 제네릭 타입으로 관리하게 되면 오히려 더 코드가 복잡해질까요??

Reply:
제네릭 사용 예시를 적어주실 수 있을까요?
아니면 Enum으로 message를 관리할 수도 있을 것 같아요!

Reply:
```suggestion
	public ApiResponse(HttpStatus status, T message, T data) {
		this.code = status.value();
		this.status = status;
		this.message = message;
		this.data = data;
	}
```

제 생각도 이런 식으로 바꾸고, 말씀하신 대로 enum으로 관리하는 방법이에요!
혹시 코드가 복잡해진다면 String으로 해도 좋을 것 같아요 :)

Reply:
음 그러면 `ResponseMessage`라는 추상 클래스를 하나 만들어서 그곳에 상수로 Message를 추가하도록 할까요? 그러면 제네릭 타입 검사는 없어도 될 것 같아요!

Reply:
너무 좋습니다!! 감사합니다

Reply:
```suggestion
public abstract class ResponseMessage {

	public static final String COMMENT_SAVE_SUCCESS = "댓글 등록에 성공했습니다.";

}

@ResponseStatus(HttpStatus.CREATED)
@PostMapping
public ApiResponse<CommentSaveResponse> saveComment(@Valid @RequestBody CommentSaveRequest request) {
	LocalDateTime createdAt = LocalDateTime.now();
	return ApiResponse.of(HttpStatus.CREATED, ResponseMessage.COMMENT_SAVE_SUCCESS,
		commentService.save(request.toServiceRequest(), createdAt));
}
```

Reply:
요런 식으로 수정하겠습니다! 피드백 감사합니당